compiler/src/iree/compiler/Codegen/Dialect/GPU/Transforms/FuseAndHoistParallelLoops.cpp

```diff
@@ -113,15 +114,63 @@ struct FuseTileableDestinationProducers final
   }
 };
 
+struct FuseTilableForallConsumers final
```

Comment:
It looks like this pass is kind of "tile and fuse everything we can into 1 loop" now. Maybe it should be renamed as well. Perhaps something like `TileAndFuseToCombinedParallelLoops`?

Reply:
After some offline discussion I think I will split this pattern out of this pass and run it separately. I'd like to do it in a follow up though to avoid further complicating this PR, but for now I'll add a TODO here and keep the pass name the same for that reason.